### DISCUSSION TITLE
DagRun creation

### DISCUSSION THREAD

**Discussion by kundan13kumar2** [apache/airflow#22523] [Q&A]
Is there any way to create dagRun with microsecond precision? I am using MWAA with airflow version v2.0.2 and sending execution time with microsecond. But After creation of dagRun, MWAA truncate microseconds and return the dagRunId with seconds only. I suppose "replace_microseconds"=false is default behaviour and it shouldn't truncate microsecond. 
Also to trigger my dagRun, i am sending some other configuration also.
Note: I am using dags trigger cli.

**Comment by potiuk** [comment]
I think you should look at migrating to Airflow 2.2 (supported by MWAA already). There, run_id is the key not execution date and you can specify whatever run_id you want.

**Reply by kundan13kumar2** [reply]
@potiuk ok.
But also there is unique constraints 
UniqueConstraint('dag_id', 'execution_date'), 
 UniqueConstraint('dag_id', 'run_id'), 

So, when I am sending same execution date with different dagRunId , throws the exception.
So, point is , can't we send our execution_date with milliseconds?
Also, When i am sending, it is getting truncated to seconds.

**Reply by potiuk** [reply]
Not in 2.2

**Reply by potiuk** [reply]
Just migrate to 2.2 . Your problem will be solved. I have NO IDEA if there is some truncation in MWAA - if you want to check, you can raise support question there rather than here. But I know that in Airflow 2.2 the community changed the key to use run id and does not use execution date as key any more.

So either  ask in MWAA forums if you want to see what you can do pre 2.2 or  migrate to 2.2 :)
